demofile-parsing/Round.js

```diff
@@ -94,6 +94,38 @@ class Round {
     var [startB, endB] = timeB;
     return ((startB <= endA) || (endA >= startB));
   }
+
+  getKills() {
+
+    // Keeping a ledger of the name of the killer for each kill in the round.
+    var roundKillerNames = [];
+    for (var i = 0; i < this.keyEvents.length; i++) {
+
+      // If the event is a player death add the killer to the array.
+      if (this.keyEvents[i].type == 'player_death') {
+        console.log(this.keyEvents[i]);
+        roundKillerNames.push(this.keyEvents[i].attacker_name);
```

Comment:
Is there a reason to make a killerNames array? In the next function we convert this array to a hash map. I think we could do this in one step without using an array. 🤔 💭 

Maybe we could have a class variable called `playerKills = {playername: killcount}

Reply:
This might well do the trick for now. In principle though, this seems more like a class variable for an eventual 'Player.js' class we might have to look into. 

In any case - I think for sure the array to hash map is a bit redundant indeed. Let me cook up smth. Ossssu <3

Reply:
@tjlam @Eldrick19 I added a commented out implementation of playerKills as a class variable. Can't get it to work.

I'm running into async hell when it tries to access keyAvents at the same time from two functions. Leaving it as is for the moment.